logagg/collector.py

```diff
@@ -61,6 +62,15 @@ def __init__(self, fpaths, nsq_sender,
         self.formatters = {}
         self.queue = Queue.Queue(maxsize=self.QUEUE_MAX_SIZE)
 
+    def validate_log_format(self, log):
+        for key in log:
+            #To avoid duplicate information
+            if key in log and key in log['data']:
+                log[key] = log['data'].pop(key)
+
+            assert bool(key in self.LOG_STRUCTURE)
+            assert isinstance(log[key], self.LOG_STRUCTURE[key])
+
```

Comment:
- Move redundancy removal to a separate function
- bool() not required
- we need to test whether value is empty or not (are all values required to be non empty?)

Reply:
@prashanthellina  I was about to make _remove_redundancy_remove a seperate function but then I thought why do a separate iteration on the keys if we can complete it in one..?